lib/oli_web/controllers/cognito_controller.ex

```diff
@@ -0,0 +1,151 @@
+defmodule OliWeb.CognitoController do
+  use OliWeb, :controller
+  import Oli.HTTP
+  import Oli.Utils
+  import Ecto
+  import Ecto.Changeset
+  require Logger
+
+  alias Oli.Repo
+  alias Oli.Accounts
+  alias Oli.Groups.CommunityAccount
+  alias Oli.Institutions.SsoJwks
+
+  def launch(conn, params) do
+    try do
+      with {:ok, valid_params} <- validate_query_params(params) do
+        jwk = get_jwk(Map.get(params, "id_token"))
+        jwk_from_pem = JOSE.JWK.from_pem(jwk.pem)
+        case JOSE.JWT.verify_strict(jwk_from_pem, ["RS256"], valid_params.id_token) do
+          {true, verified_jwt, _} ->
+            case Accounts.insert_or_update_lms_user(%{
+                  sub: Map.get(verified_jwt.fields, "sub"),
+                  preferred_username: Map.get(verified_jwt.fields, "cognito:username"),
+                  email: Map.get(verified_jwt.fields, "email"),
+                  can_create_sections: true
+                }) do
+              {:ok, user} ->
+                  attrs = %{user_id: user.id, community_id: valid_params.community_id}
+
+                  %CommunityAccount{}
+                  |> CommunityAccount.changeset(attrs)
+                  |> Repo.insert()
```

Comment:
We could use `Oli.Groups.create_community_account(attrs)` method that does basically the same